tests/jsmarionette/runner/marionette-js-runner/host/lib/request.js

```diff
@@ -71,6 +77,13 @@ module.exports = function request(socketPath, path, json, retry) {
           err.stack = json.stack;
           reject(err);
         } else {
+          // start_runner is special and troublesome.
```

Comment:
Sure thing, when we get a timeout, we ended up returning an empty response which would resolve the promise, not reject it. To surface the actual timeout error, I opted to only resolve when json data is present (which contains the ID of the runner).

In hindsight, this isn't the best solution. I should instead reject the promise when I get a response with no data. I'll adjust this accordingly in the next revision.